src/main/java/service/GameService.java

```diff
@@ -0,0 +1,147 @@
+package service;
+
+import dao.GameRoomDao;
+import dao.GameRoomDto;
+import dao.PieceDao;
+import dao.converter.BoardConverter;
+import dao.init.ConnectionGenerator;
+import domain.JanggiGame;
+import domain.board.Board;
+import domain.board.BoardFactory;
+import domain.board.pathfinder.DefaultPathFinderFactory;
+import domain.piece.Piece;
+import domain.piece.character.Team;
+import domain.point.Point;
+import java.sql.Connection;
+import java.util.Map;
+import java.util.Optional;
+import queue.MessageQueue;
+import view.SangMaOrderCommand;
+
+public class GameService {
+
+    private final GameRoomDao gameRoomDao;
+    private final PieceDao pieceDao;
+    private final ConnectionGenerator connectionGenerator;
+
+    private JanggiGame janggiGame;
+
+    public GameService(GameRoomDao gameRoomDao, PieceDao pieceDao, ConnectionGenerator connectionGenerator) {
+        this.gameRoomDao = gameRoomDao;
+        this.pieceDao = pieceDao;
+        this.connectionGenerator = connectionGenerator;
+    }
+
+    public void executeDelayedQueries() {
+        MessageQueue.getInstance().executeDelayedQueries(getConnection());
+    }
+
+    public boolean existsGameRoom(final String gameRoomName) {
+        try {
+            final Optional<GameRoomDto> gameRoomDto = gameRoomDao.findByName(getConnection(), gameRoomName);
+            return gameRoomDto.isPresent();
+        } catch (RuntimeException e) {
+            return false;
+        }
+    }
+
+    public void loadGame(String gameRoomName) {
+        janggiGame = loadGameByGameRoomName(gameRoomName);
+    }
+
+    public void createNewGame(String gameRoomName,
+                              BoardFactory boardFactory,
+                              SangMaOrderCommand choSangMaOrderCommand,
+                              SangMaOrderCommand hanSangMaOrderCommand) {
+        final Team firstTurn = Team.CHO;
+        final JanggiGame newGame = new JanggiGame(
+                gameRoomName,
+                boardFactory.createBoard(
+                        DefaultPathFinderFactory.getInstance(), choSangMaOrderCommand, hanSangMaOrderCommand
+                ),
+                firstTurn
+        );
+        gameRoomDao.insert(new GameRoomDto(null, gameRoomName, firstTurn));
+        pieceDao.insertAll(BoardConverter.convertToPieceDtos(newGame.getPieceByPoint(), gameRoomName));
+
+        executeDelayedQueries();
+        janggiGame = newGame;
+    }
+
+    public void movePiece(final Point source, final Point destination) {
+        getGameOrThrow().movePiece(source, destination);
```

Comment:
janggiGame 객체에게 `movePice` 메시지를 보낸 부분이었는데, `getGameOrThrow()`의 결과가 JanggiGame 객체인지 한눈에 잘 안와닿는 것 같아요. 👀

아래와 같이 `getGameOrThrow()`를 변수로 추출하여 JanggiGame 객체임을 조금 더 드러내주시면 좋을 것 같아요. 
메서드가 체이닝되어 호출되니 의도가 잘 드러나지 않는 것 같아요. 🤔 

```suggestion
        JanggiGame janggiGame = getGameOrThrow();
        janggiGame.movePiece(source, destination);
```